File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/clinical/eye-examination/shared/AblationProfile.model.ts
import {
  NgForm,
  FormGroup,
  FormControl,
  Validators,
  FormBuilder,
  ReactiveFormsModule
} from '@angular/forms';
import * as moment from 'moment/moment';

export class AblationProfileModel {
  public Id: number = 0;
  public MasterId: number = 0;
  public Profile: string = null;
  public DZPFactor: string = null;
  public isPTKPerformed: boolean = false;
  public isSXDone: boolean = false;
  public CreatedBy: number = 0;
  public CreatedOn: Date = new Date() ;
  public IsOD: boolean ;
  public Notes: string = null;

  constructor() {
    
  }
  dateValidator(control: FormControl): { [key: string]: boolean } {
    var currDate = moment().format('YYYY-MM-DD HH:mm');
    if (control.value) { // gets empty string for invalid date such as 30th Feb or 31st Nov)
      if ((moment(control.value).diff(currDate) > 0)
        || (moment(currDate).diff(control.value, 'years') > 200)) //can select date upto 200 year past from today.
        return { 'wrongDate': true };
    }
    else
      return { 'wrongDate': true };
  }

}

File: Code/Components/DanpheEMR.ServerModel/ClinicalModels/ClinicalEye/AblationProfileModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class AblationProfileModel
    {
        [Key]
        public int Id { get; set; }
        public int MasterId { get; set; }
        public string Profile { get; set; }
        public string DZPFactor { get; set; }
        public Boolean isSXDone { get; set; }
        public Boolean isPTKPerformed { get; set; }
        public int CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public Boolean IsOD { get; set; }
        public string Notes { get; set; }
    }
}

